src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Orchestration.cs

```diff
@@ -35,13 +36,18 @@ public partial class ReindexDurableFunction
         [OrchestrationTrigger] IDurableOrchestrationContext context,
         ILogger logger)
     {
-        EnsureArg.IsNotNull(context, nameof(context));
+        // The ID should be a GUID as generated by the trigger, but we'll assert here just to make sure!
+        EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
 
         logger = context.CreateReplaySafeLogger(logger);
-        ReindexInput input = context.GetInput<ReindexInput>();
+        ReindexCheckpoint input = context.GetInput<ReindexCheckpoint>();
 
-        // The ID should be a GUID as generated by the trigger, but we'll assert here just to make sure!
-        context.ThrowIfInvalidOperationId();
+        // Backfill batching options
+        input.Batching ??= new BatchingOptions
+        {
+            MaxParallelCount = _options.MaxParallelBatches,
```

Comment:
should probably deprecate the function options and instead rely on input

Reply:
Why we prefer moving this setting to input?

The benefit I could think about is to better optimize based on input? 

The drawback I could think about is we could not globally change these value by deployment -- like last time we reduced parallel level to reduce CPU usage.

Reply:
We have to set this information in the input, as it impacts the number of generated activities. However, the number of threads does not need to be based as input

Reply:
so  for different reindex job, we will have different parallel level in the future?